### ISSUE TITLE
Evaluation Metric Clarification

### ISSUE THREAD

**Issue by Fardeen786-eng** [guosyjlu/DS-Agent#12]
Thank you for your insightful work and contribution. I just have a few questions regarding the work in the paper and the code implementation:
1. I understand that we have used the training data or the available dataset with target values for our development stage to better understand which codes work through an iterative process. However, for some competitions, we have test sets available, and we can even make submissions to Kaggle to get real leaderboard scores. Wouldn't this method of testing be more beneficial and competitive?
2. In the paper, Tables 1 and 3 show the mean rank and best rank information for the datasets. I didn't fully understand the meaning of this ranking. Could you provide more insights into the evaluation and the criteria used?
3. In our code base, we provide train.py as a starter file, and I noticed that we perform a few preprocessing steps which are crucial to the ML life cycle. Would it be correct to do this, as generation from scratch tends to produce more hallucinations and mistakes? This in-filling method adds human cost to creating these files. Is this a good way to measure our system's performance?

**Comment by guosyjlu** [comment]
Hi, thanks for your kind words!
1. Yes, of course. Note that this is a research paper; thus, we simulate this process via predefined train/valid/test splits. If you want to adapt DS-Agent in real-world Kaggle competitions, it is better to utilize Kaggle API to get real leaderboard scores for testing.
2. Assume there are two agents A and B. Given a data science task, both agents perform 5 random trials to build models. Then, we use the predefined evaluation metric to evaluate the built model in the testing set. As such, we can rank these ten built models via the evaluation results.
> Assume the models built by Agent A attains the rank [1,3,5,7,9], and the models built by Agent B attains the rank [2,4,6,8,10].
As such, MeanRank(A)=mean([1,3,5,7,9])=5, BestRank(A)=min([1,3,5,7,9])=1. Similarly, MeanRank(B)=6, BestRank(B)=2.
3. This is a good question! Since we focus on building machine learning models via LLM agent in this paper, we simplify the data engineering part of automated data science by providing an initial python script. However, this part can also be fully automated by another well-crafted LLM agent to further decrease the human labor. I believe this is a simpler task for LLMs, compared to the complex model building and exploration.
